components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/util/EndpointUtil.java

```diff
@@ -221,17 +222,22 @@ public static String getUserInfoClaimDialect() {
         if (authorizationHeader == null) {
             throw new OAuthClientException("Authorization header value is null");
         }
+        String errMsg = "Error decoding authorization header. Space delimited \"<authMethod> <base64encoded" +
+                "(username:password)>\" format violated.";
         String[] splitValues = authorizationHeader.trim().split(" ");
         if (splitValues.length == 2) {
-            byte[] decodedBytes = Base64Utils.decode(splitValues[1].trim());
-            String userNamePassword = new String(decodedBytes, Charsets.UTF_8);
-            String[] credentials = userNamePassword.split(":");
-            if (credentials.length == 2) {
-                return credentials;
+            if (HTTP_REQ_HEADER_AUTH_METHOD_BASIC.equals(splitValues[0])) {
```

Comment:
@ayshsandu : I don't think this fix is correct for public branch. Did you check whether you can reproduce this issue in master branch. This might be already fixed in master branch with the pluggable authenticators we introduced.

Reply:
Yes, with the pluggable authenticators the wso2/product-is#3321 doesn't get reproduced. However, I fixed this as this is not correct in the code.